testing/regress/ecl/parquetTypes.ecl

```diff
@@ -0,0 +1,617 @@
+
+// ======================== UTF8 ========================
+
+utf8Record := {UNSIGNED testid, STRING3 testname, UTF8 value};
+
+utf8DatasetOut := DATASET([
+    {090, 'abu', U'UTF8_Normal'},
+    {091, 'abv', U'UTF8_测试'},
+    {092, 'abw', U''}
+], utf8Record);
+
+ParquetIO.Write(utf8DatasetOut, dropzoneDirectory + '/UTF8Test.parquet', TRUE);
+
+utf8DatasetIn := ParquetIO.Read(utf8Record, dropzoneDirectory + '/UTF8Test.parquet');
+
+{UNSIGNED testid, STRING3 testname, BOOLEAN isEqual} utf8Join (utf8DatasetOut a, utf8DatasetIn b) := TRANSFORM
+    SELF.testid := a.testid;
+    SELF.testname := a.testname;
+    SELF.isEqual := a.value = b.value;
+END;
+
+utf8CompareResult := JOIN(utf8DatasetOut, utf8DatasetIn, LEFT.testid = RIGHT.testid, utf8Join(LEFT, RIGHT), ALL);
+
+utf8Result := IF(COUNT(utf8CompareResult(isEqual = FALSE)) = 0, 'Pass', 'Fail: UTF8 data mismatch');
+
+// ======================== UNICODE ========================
+
+unicodeRecord := {UNSIGNED testid, STRING3 testname, UNICODE value};
+
+unicodeDatasetOut := DATASET([
+    {100, 'acb', U'Unicode1'},
+    {101, 'acc', U'Unicode2'},
+    {102, 'acf', U'Unicode5'}
+], unicodeRecord);
+
+ParquetIO.Write(unicodeDatasetOut, dropzoneDirectory + '/UnicodeTest.parquet', TRUE);
+
+unicodeDatasetIn := ParquetIO.Read(unicodeRecord, dropzoneDirectory + '/UnicodeTest.parquet');
+
+{UNSIGNED testid, STRING3 testname, BOOLEAN isEqual} unicodeJoin (unicodeDatasetOut a, unicodeDatasetIn b) := TRANSFORM
+    SELF.testid := a.testid;
+    SELF.testname := a.testname;
+    SELF.isEqual := a.value = b.value;
+END;
+
+unicodeCompareResult := JOIN(unicodeDatasetOut, unicodeDatasetIn, LEFT.testid = RIGHT.testid, unicodeJoin(LEFT, RIGHT), ALL);
+
+unicodeResult := IF(COUNT(unicodeCompareResult(isEqual = FALSE)) = 0, 'Pass', 'Fail: UNICODE data mismatch');
+
+// ======================== SET OF INTEGER ========================
+
+SetOfIntegerRecord := {UNSIGNED testid, STRING3 testname, SET OF INTEGER value};
+
+setOfIntegerDatasetOut := DATASET([
+    {110, 'acg', [1, 2, 3]},
+    {113, 'acj', [10, 11, 12]},
+    {114, 'ack', [13, 14, 15]}
+], SetOfIntegerRecord);
+
+ParquetIO.Write(setOfIntegerDatasetOut, dropzoneDirectory + '/SetOfIntegerTest.parquet', TRUE);
+
+setOfIntegerDatasetIn := ParquetIO.Read(SetOfIntegerRecord, dropzoneDirectory + '/SetOfIntegerTest.parquet');
+
+{UNSIGNED testid, STRING3 testname, BOOLEAN isEqual} setOfIntegerJoin (setOfIntegerDatasetOut a, setOfIntegerDatasetIn b) := TRANSFORM
+    SELF.testid := a.testid;
+    SELF.testname := a.testname;
+    SELF.isEqual := a.value = b.value; 
+END;
+
+setOfIntegerCompareResult := JOIN(setOfIntegerDatasetOut, setOfIntegerDatasetIn, LEFT.testid = RIGHT.testid, setOfIntegerJoin(LEFT, RIGHT), ALL);
+
+setOfIntegerResult := IF(COUNT(setOfIntegerCompareResult(isEqual = FALSE)) = 0, 'Pass', 'Fail: SET OF INTEGER data mismatch');
+
+// ======================== REAL8 ========================
```

Comment:
Rather than create a new section for each type variation you should just add an additional column to the existing REAL dataset on line 92.